execution/gethexec/sequencer.go

```diff
@@ -758,6 +768,57 @@ func (s *Sequencer) precheckNonces(queueItems []txQueueItem) []txQueueItem {
 	return outputQueueItems
 }
 
+func deleteFiles(files ...*os.File) {
+	for _, f := range files {
+		if err := os.Remove(f.Name()); err != nil {
+			log.Error("Error removing file", "name", f.Name())
+		}
+	}
+}
+
+func closeFiles(files ...*os.File) {
+	for _, f := range files {
+		if err := os.Remove(f.Name()); err != nil {
+			log.Error("Error closing file", "name", f.Name())
+		}
+	}
+}
+
+// createBlockWithProfiling runs create block with tracing and CPU profiling
+// enabled. If the block creation takes longer than 5 seconds, it keeps both
+// and prints out filenames in an error log line.
+func (s *Sequencer) createBlockWithProfiling(ctx context.Context) bool {
+	id := uuid.NewString()
+	pprofFile, err := os.CreateTemp("", id+".pprof")
+	if err != nil {
+		log.Error("Creating temporary file for profiling CPU", "error", err)
+	}
+	traceFile, err := os.CreateTemp("", id+".trace")
+	if err != nil {
+		log.Error("Creating temporary file for tracing", "error", err)
+	}
+	if err := pprof.StartCPUProfile(pprofFile); err != nil {
+		log.Error("Starting CPU profiling", "error", err)
+		deleteFiles(pprofFile)
+	}
+	if err := trace.Start(traceFile); err != nil {
+		deleteFiles(traceFile)
+		log.Error("Starting tracing", "error", err)
+	}
+	start := time.Now()
+	res := s.createBlock(ctx)
+	elapsed := time.Since(start)
+	pprof.StopCPUProfile()
+	trace.Stop()
```

Comment:
In case createBlock panics, can we stop the trace / pprof and log the file paths?

Reply:
I think now only `debug.Stack()` is printed out in the deferred function in createBlock. 
https://github.com/OffchainLabs/nitro/blob/3b33bc4f74356e95bdf98d51ebd3e3df09fe92df/execution/gethexec/sequencer.go#L826-L839

I am not sure if pprof is useful in case of panic, but if it is then we probably should call `pprof.StopCPUProfile` and `trace.Stop()` + log the tmp path in another deferred recover in `createBlockWithProfiling`. If pprof is not useful, then all is good - we'd have a trace in the logs.

Reply:
Discussed offline.